scripts/deploy-gh-pages.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+mkdir tmp
+cd tmp
+
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+# Pull requests and commits to other branches shouldn't try to deploy, just build to verify
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o -z "$TRAVIS_TAG" ]; then
+    echo "Skipping deploy"
+    exit 0
+fi
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+REPO_WITH_TOKEN=${REPO/https:\/\/github.com\//https://${GH_TOKEN}@github.com/}
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deply)
+git clone $REPO src
+cd src; git checkout $SOURCE_BRANCH
```

Comment:
I don't feel the necessity of `;`, why not remove this and write two separate commands?